docs/admin/setup/pyris.rst

```diff
@@ -31,70 +25,214 @@ E.g.: ``git clone https://github.com/ls1intum/Pyris.git Pyris``
+      name: GPT 3.5 Turbo
+      type: openai_chat
+
+Example Azure OpenAI
+********************
+.. code-block:: yaml
 
-- ``PYRIS_DOCKER_TAG``: The Docker tag to use. Ideally use ``latest`` here.
-- ``PYRIS_APPLICATION_YML_FILE``: The absolute path to the ``application.yml`` file that you created. E.g.: ``/opt/pyris/application.yml``
-- ``NGINX_PROXY_SSL_CERTIFICATE_PATH``: The absolute path to the SSL certificate to use. Defaults to self-signed certificates. For production, you should use a valid SSL certificate. For local development, you an ignore this variable.
-- ``NGINX_PROXY_SSL_CERTIFICATE_KEY_PATH``: The absolute path to the SSL certificate key to use. Defaults to self-signed certificates. For production, you should use a valid SSL certificate. For local development, you an ignore this variable.
+      api_key: "<your_azure_api_key>"
+      tools: []
+      capabilities:
+        input_cost: 6
+        output_cost: 16
+        gpt_version_equivalent: 4.5 # This is the equivalent GPT version of the model. We use 4.5 for GPT 4 Omni model.
+        context_length: 128000
+        vendor: "OpenAI"
+        privacy_compliance: True
+        self_hosted: False
+        image_recognition: True
+        json_mode: True
+      description: "GPT 4 Omni on Azure"
+      id: "azure-gpt-4-omni"
+      name: "GPT 4 Omni"
+      type: "azure_chat"
+      endpoint: <your_azure_model_endpoint>
+      api_version: "2024-02-15-preview"
+      azure_deployment: "gpt4o"
+      model: "gpt4o"
 
 Start Pyris
 """""""""""
+Using local environment
+***********************
+
+.. warning::
+    For local Weaviate vector database setup, please refer to `Weaviate Docs <https://weaviate.io/developers/weaviate/quickstart>`_.
+
+=============
+Prerequisites
+=============
+- Clone the Pyris repository to your local machine.
+- Ensure you correctly configured ``llm_config.local.yml`` file.
+- Ensure you correctly configured ``application.local.yml`` file.
+
+=======================================================
+Setup instructions
+=======================================================
+Follow the following steps for the local Pyris setup:
+
+1. **Check python version:** python --version (should be 3.12)
+2. **Install packages:** pip install -r requirements.txt
+3. **Start Pyris** using the following command:
+    .. code-block:: bash
+
+      APPLICATION_YML_PATH=<path-to-your-application-yml-file> LLM_CONFIG_PATH=<path-to-your-llm-config-yml> uvicorn app.main:app --reload
+
+4. **You can now access the API docs under the following link:** http://localhost:8000/docs
+
+This setup should help you run the Pyris application on your local machine.
+Ensure you modify the configuration files as per your specific requirements before deploying.
+
+Using Docker
+************
+
+You can run Pyris in different environments: ``development`` or ``production``. Docker Compose is used to orchestrate the different services, including ``Pyris``, ``Weaviate``, and ``Nginx``.
```

Comment:
Why is nginx needed?

Reply:
@Hialus

Reply:
I'm not sure about the exact reasoning behind using Nginx, but I assume it is configured that way so that we can deploy Pyris to a different GPU server if we use e.g. `Ollama` for the open source models in the future.

Reply:
I have Pyris configured in the same virtual network as Artemis. Which could be the use case for most deployments . Even for development you don't need an nginx . Thus I'd remove it completely to ease the documentation . Otherwise you should explain in detail why nginx is needed .

Reply:
Ok, thank you for the thorough review. Today, I will address the other issues you raised as well. Since I was not involved in the docker and Nginx setup, I might not provide a reasonable explanation as to why Nginx is needed in the current version of the documentation. @Hialus can help us better, when he is back.

Reply:
Just as a hint: This is the excerpt from the configuration on our test system:

```
  pyris:
    image: ghcr.io/ls1intum/pyris:latest #:1.0.1
    restart: unless-stopped
    environment:
      APPLICATION_YML_PATH: "/config/application.yml"
      LLM_CONFIG_PATH: "/config/llm_config.yml"
    volumes:
      - ./data/pyris:/config
    expose:
      - "8000"
    ports:
      - "8000:8000"
    networks:
      - artemis-net
    command: ["uvicorn", "app.main:app", "--host", "0.0.0.0", "--port", "8000"]

  weaviate:
    image: cr.weaviate.io/semitechnologies/weaviate:1.25.3
    restart: unless-stopped
    command: --host 0.0.0.0 --port 8001 --scheme http
    expose:
      - 8001
      - 50051
    ports:
      - 8001:8001
      - 50051:50051
    volumes:
      - ./data/weaviate:/var/lib/weaviate
    environment:
      - LOG_LEVEL=debug
      - QUERY_DEFAULTS_LIMIT=25
      - AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED=true
      - PERSISTENCE_DATA_PATH=/var/lib/weaviate
      - DEFAULT_VECTORIZER_MODULE=none
      - ENABLE_MODULES=
      - CLUSTER_HOSTNAME=node1
      - LIMIT_RESOURCES=true
      - DISK_USE_WARNING_PERCENTAGE=80
      - vectorCacheMaxObjects=1000000
    networks:
      - artemis-net
```

Reply:
Thank you for providing this docker-xompose file. I will make an additional PR on the Pyris side to add a configuration without Nginx.

Reply:
Regarding Nginx, please see: https://github.com/ls1intum/Pyris/pull/159